### DISCUSSION TITLE
Export control implementation

### DISCUSSION THREAD

**Discussion by raenji-sk** [wills106/homeassistant-solax-modbus#213] [Ideas]
Back to my original thought. Would something like this be doable with the intgegration?

https://github.com/squishykid/solax/issues/58

The guy is using X1 mini, not sure if it is applicable to other ongrid models but maybe it's worth a try?

Except I'd need replacing the SDM230 meter with something else (Shell 3em in my case) get the data from it and send it via Modbus in regular intervals to the X3 MIC G2 to control export levels. So basically a simulated power meter.

**Comment by wills106** [comment]
I think it's technically something that would be possible.

But it's not something I would know how to implement at the moment. The Integration would need to be able read sensors from another Integration, which I don't know how to do.

I also don't know if I could send two sets of Modbus Instructions from a single Waveshare. Or if you would need two separate Waveshares? One for the Integration and one to pretend to be a Modbus Meter.

**Reply by raenji-sk** [reply]
Reading a sensor like a Shelly would probably be not a problem. Then just convert the measured value to hex and calculate CRC and send it via modbus to inverter.
But yes I agree it is a bit of a stretch.
Perhaps it could be doable via node red and modbus slave server. Tried googling some solution but ended up with nothing...
Not that I can't live without it. But perhaps at some point it could be done.
If you ever feel like trying it and need a tester then let me know.
